File: notes/1248-Fall2024/155E/code/wins.c
/**
 * TODO: full author header
 *
 */
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <math.h>

int main(int argc, char **argv) {

    if(argc != 3) {
        printf("ERROR: provide two command line arguments (points for, against)\n");
        exit(1);
    }

    int pointsFor = atoi(argv[1]);
    int pointsAgainst = atoi(argv[2]);

    double winPercentage = pow(pointsFor, 2.37) / (pow(pointsFor, 2.37) + pow(pointsAgainst, 2.37));

    int numWins = round(winPercentage * 12);
    int numLoss = 12 - numWins;

    printf("Win Percentage: %.2f%%\n", winPercentage * 100.0);
    printf("Win/Loss: %d/%d\n", numWins, numLoss);


    return 0;
}

File: notes/1248-Fall2024/155H/code/Wins.java
package unl.soc;

/**
 * TODO:
 */
public class Wins {

	public static final double EXPONENT = 2.37;

	public static void main(String args[]) {
		// TODO: basic error checking

		int pointsFor = Integer.parseInt(args[0]);
		int pointsAgainst = Integer.parseInt(args[1]);

		double winPercentage = Math.pow(pointsFor, EXPONENT)
				/ (Math.pow(pointsFor, EXPONENT) + Math.pow(pointsAgainst, EXPONENT));

		int wins = (int) Math.round(winPercentage * 12);
		int loss = 12 - wins;

		System.out.printf("Win Percentage: %f%%\n", winPercentage * 100);
		System.out.printf("Win/Loss: %d/%d\n", wins, loss);

		String format = """
				Win Percentage: %f%%
				Win/Loss: %d/%d
				""";
		System.out.printf(format, winPercentage * 100, wins, loss);

	}

}
